### PR TITLE
Add .NET 5 to Elastic.Apm.Tests

### LINKED ISSUE #962: [CI] Add .NET 5 support in CI
.NET 5 is already RC - can be found [here](https://dotnet.microsoft.com/download/dotnet/5.0) including the [install scripts](https://dotnet.microsoft.com/download/dotnet-core/scripts).

We should have that in CI. Once that's done I'll add .NET 5 as target for all our tests and sample apps, so all our tests run on the new version.

So the point of this issue: make sure it's installed in CI, once that's done I'll make sure the tests also utilize it.

### PR DESCRIPTION
**Pull request by russcam**
This commit adds .NET 5 to the Elastic.Apm.Tests.

- global.json sdk version updated to 5.0.100
- Dockerfile updated to use mcr.microsoft.com/dotnet/sdk:5.0 image
- scripts updated to install net 5.0 sdks and Visual Studio tooling that supports .net 5.0

Closes #962

### CODE REVIEW

**Comment by russcam** [comment]
Installing tools fails

> [2020-11-16T02:16:56.277Z] visualstudio2019buildtools has been installed. 
> [2020-11-16T02:16:56.277Z]   visualstudio2019buildtools may be able to be automatically uninstalled.
> [2020-11-16T02:16:56.277Z]  The install of visualstudio2019buildtools was successful.
> [2020-11-16T02:16:56.277Z]   Software installed to 'C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools'
> [2020-11-16T02:16:56.277Z] 
> [2020-11-16T02:16:56.277Z] Chocolatey installed 1/1 packages. 
> [2020-11-16T02:16:56.277Z]  See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
> [2020-11-16T02:16:56.277Z] visualstudio2019buildtools installation failed. See the logs.
> 
> script returned exit code 1

Looks like the installation is successful, though a non-zero exit code is returned. Might be good to emit the chocolatey logs when this happens?

File: .ci/windows/index.html [hunk 1]
```diff
@@ -0,0 +1 @@
...tr><tr><td colspan=2>TestFleetGateway/send_no_event_and_receive_no_action</td><td>failed 2 times:</td></tr><tr><td> </td><td><a href=https://beats-ci.elastic.co/job/Beats/job/beats/job/7.9/260/console rel=nofollow>elastic&#43;beats</a></td><td>failed 2 times</td></tr><tr><td colspan=2>TestIncludedExcludedFiles</td><td>failed 2 times:</td></tr><tr><td> </td><td><a href=https://beats-ci.elastic.co/job/Beats/job/beats/job/master/706/console rel=nofollow>elastic&#43;beats</a></td><td>failed 2 times</td></tr><tr><td colspan=2>test_base</td><td>failed 2 times:</td></tr><tr><td> </td><td><a href=https://beats-ci.elastic.co/job/Beats/job/beats/job/master/706/console rel=nofollow>elastic&#43;beats</a></td><td>failed 2 times</td></tr><tr><td colspan=2>test_close_renamed</td><td>failed 2 times:</td></tr><tr><td> </td><td><a href=https://beats-ci.elastic.co/job/Beats/job/beats/job/7.x/594/console rel=nofollow>elastic&#43;beats</a></td><td>failed 2 times</td></tr><tr><td colspan=2>ExampleWrapper</td><td>failed 1 times:</td></tr><tr><td> </td><td><a href=https://beats-ci.elastic.co/job/Beats/job/beats/job/7.x/613/console rel=nofollow>elastic&#43;beats</a></td><td>failed 1 times</td></tr></table>
```
Comment (gregkalapos) [inline .ci/windows/index.html:1]: Can we remove this? Seems like a leftover.

**Review by gregkalapos — COMMENTED** [review]
Seems we still hit the `RazorTagHelper` issue.

```
 C:\Program Files\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Microsoft.NET.Sdk.Razor.CodeGeneration.targets(97,5): error MSB4018: The "RazorTagHelper" task failed unexpectedly. 
```

File: .ci/windows/index.html (hunk 1 above)
Comment (russcam) [reply .ci/windows/index.html:1]: Yep - looks like it crept in https://github.com/elastic/apm-agent-dotnet/commit/85295009ccadf73b8501200a2810526ace58f26d

**Comment by russcam** [comment]
Everything looks to build now, but looks like there was an intermittent failing test

**Comment by russcam** [comment]
Jenkins run the tests please

**Comment by russcam** [comment]
Looks like the only thing that has failed is dotnet testing in apm integration testing. Will open a PR for this in https://github.com/elastic/apm-integration-testing

**Comment by russcam** [comment]
jenkins run the tests please

**Comment by russcam** [comment]
jenkins run the tests please

**Comment by gregkalapos** [comment]
Test `MetricsTests.Elastic.Apm.Tests.MetricsTests.ProviderWithException` failed with:


```
Error
Expected testLogger.Lines[1] "[2020-12-21 02:58:39.191 +00:00][Warning] - {SystemTotalCpuProvider} TraceEventSession initialization failed - GC metrics won't be collected" to contain "Failed reading test metric 1 times".
```

That log is totally unrelated and adapted the test to be more dynamic. With that hopefully the test will be green.

File: .ci/windows/index.html (hunk 1 above)
Comment (gregkalapos) [reply .ci/windows/index.html]: I removed it.

**Review by gregkalapos — APPROVED** [review]
I think this is ok to merge.

Integration test failed, but that's already the case regardless of this PR, so likely unrelated.

**Pull request merged**